ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java

```diff
@@ -1176,16 +1206,19 @@ private void verifyPutFailure(MockId idToPut, StoreErrorCodes expectedErrorCode)
    * @param idToDelete the {@link MockId} to DELETE.
    * @param expectedErrorCode the expected {@link StoreErrorCodes} for the failure.
    */
-  private void verifyDeleteFailure(MockId idToDelete, StoreErrorCodes expectedErrorCode) {
-    MessageInfo info = new MessageInfo(idToDelete, DELETE_RECORD_SIZE, Utils.getRandomShort(TestUtils.RANDOM),
-        Utils.getRandomShort(TestUtils.RANDOM), System.currentTimeMillis());
+  private void verifyDeleteFailure(MockId idToDelete, StoreErrorCodes expectedErrorCode) throws StoreException {
+    MessageInfo info =
+        new MessageInfo(idToDelete, DELETE_RECORD_SIZE, idToDelete.getAccountId(), idToDelete.getContainerId(),
+            System.currentTimeMillis());
     MessageWriteSet writeSet =
         new MockMessageWriteSet(Collections.singletonList(info), Collections.singletonList(ByteBuffer.allocate(1)));
-    try {
-      store.delete(writeSet);
-      fail("Store DELETE should have failed");
-    } catch (StoreException e) {
-      assertEquals("Unexpected StoreErrorCode", expectedErrorCode, e.getErrorCode());
+    if (store.getStoreConfig().storeDeleteAuthorizationCheck == true) {
```

Comment:
why is this condition required?

Reply:
The result of below check depends on the value of storeDeleteAuthorizationCheck in StoreCofing.java.

Since we may change the config value, I added the if condition.